code/modules/antagonists/traitor/equipment/Malf_Modules.dm

```diff
@@ -176,71 +176,85 @@ GLOBAL_LIST_INIT(malf_modules, subtypesof(/datum/ai_module))
 	if (owner_AI.shunted)
 		return //prevent AI from activating doomsday while shunted, fucking abusers
 	active = TRUE
-	set_us_up_the_bomb(owner)
+	set_up_us_the_bomb(owner)
 
-/datum/action/innate/ai/nuke_station/proc/set_us_up_the_bomb(mob/living/owner)
-	var/pass = prob(10) ? "******" : "hunter2"
+/datum/action/innate/ai/nuke_station/proc/set_up_us_the_bomb(mob/living/owner)
 	set waitfor = FALSE
+	message_admins("[key_name_admin(owner)][ADMIN_FLW(owner)] has activated AI Doomsday.")
+	var/pass = prob(10) ? "******" : "hunter2"
 	to_chat(owner, "<span class='small boldannounce'>run -o -a 'selfdestruct'</span>")
 	sleep(5)
-	if(!owner || QDELETED(owner))
+	if(!owner || QDELETED(owner) || owner_AI.shunted)
```

Comment:
```suggestion
	if(QDELETED(owner) || owner_AI.shunted)
```
`QDELETED(owner)` already checks for `!owner`
Same for the other changes.

Reply:
`!owner` has been removed.